pkg/datasource/datasource.go

```diff
@@ -83,7 +85,25 @@ func NewOSS(ctx context.Context, c client.Client, endpoint *v1alpha1.Endpoint, o
 	return &OSS{Client: mc}, nil
 }
 
-// TODO: implement Check with specific `options`
-func (oss *OSS) Check(ctx context.Context, options any) error {
+func (oss *OSS) Check(ctx context.Context, ossInfo *v1alpha1.OSS) error {
+	if ossInfo.Bucket != "" {
+		_, err := oss.Client.BucketExists(context.Background(), ossInfo.Bucket)
+		if err != nil {
+			return err
+		}
+
+		if ossInfo.Object != "" {
+			_, err := oss.Client.StatObject(context.Background(), ossInfo.Bucket, ossInfo.Object, minio.StatObjectOptions{})
+			if err != nil {
+				// Determine whether there is a permission error
+				if minio.ToErrorResponse(err).Code == "AccessDenied" {
+					return ErrAccessPermission
+				} else {
+					return ErrAccessObject
+				}
+			}
+		}
+	}
```

Comment:
**Suggestion:** Avoid using `context.Background()` in the `Check` method of the `OSS` struct. Instead, use the `ctx` parameter that is passed to the function.
```suggestion
func (oss *OSS) Check(ctx context.Context, ossInfo *v1alpha1.OSS) error {
    if ossInfo.Bucket != "" {
        _, err := oss.Client.BucketExists(ctx, ossInfo.Bucket)
        if err != nil {
            return err
        }

        if ossInfo.Object != "" {
            _, err := oss.Client.StatObject(ctx, ossInfo.Bucket, ossInfo.Object, minio.StatObjectOptions{})
            if err != nil {
                // Determine whether there is a permission error
                if minio.ToErrorResponse(err).Code == "AccessDenied" {
                    return ErrAccessPermission
                } else {
                    return ErrAccessObject
                }
            }
        }
    }

    return nil
}
```